addons/osfstorage/models.py

```diff
@@ -196,8 +205,9 @@ def check_in_or_out(self, user, checkout, save=False):
                 self.save()
 
     def save(self):
-        self.path = ''
-        self.materialized_path = ''
+        self._path = ''
+        self._materialized_path = ''
+        self.provider = self._provider
```

Comment:
Isn't OsfStorageFileNode the only one that does **not** keep a path or materialized path?